### ISSUE TITLE
[Bug]: Update 24.0.0 beta2 to beta3, RC3 to final

### ISSUE THREAD

**Issue by RoundAboutWEB** [nextcloud/server#31891]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on Github _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

[✘] Check for expected files failed
The following extra files have been found:
    dist
    webpack.modules.js

Update failed. To resume or retry just execute the updater again.

### Steps to reproduce

1. Updateing by backend
2. Updating by CLI

Update from beta1 to beta2 was no problem

### Expected behavior

I was expecting a successful update from 24.0.0 beta2 to 24.0.0. beta3.

### Installation method

Manual installation

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 8.0

### Web server

Apache (supported)

### Database engine version

MySQL

### Are you using the Nextcloud Server Encryption module?

Encryption is Enabled

### What user-backends are you using?

- [X] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

```shell
{
    "system": {
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "cloud.email2you.de"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "24.0.0.7",
        "overwrite.cli.url": "https:\/\/cloud.email2you.de\/",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "htaccess.RewriteBase": "\/",
        "default_language": "de",
        "default_locale": "de_DE",
        "default_phone_region": "DE",
        "mail_smtpmode": "smtp",
        "mail_smtpsecure": "tls",
        "mail_smtpauthtype": "LOGIN",
        "mail_smtpauth": 1,
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "25",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "check_for_working_wellknown_setup": false,
        "skeletondirectory": "\/files\/nextcloud\/data\/eigenes-skeleton",
        "theme": "",
        "loglevel": 0,
        "maintenance": false,
        "memcache.local": "",
        "updater.release.channel": "beta",
        "updater.secret": "***REMOVED SENSITIVE VALUE***"
    }
}
```

### List of activated Apps

```shell
Enabled:
  - accessibility: 1.10.0
  - activity: 2.16.0
  - circles: 24.0.0-alpha1
  - cloud_federation_api: 1.7.0
  - comments: 1.14.0
  - contacts: 4.1.0
  - contactsinteraction: 1.5.0
  - dashboard: 7.4.0
  - dav: 1.22.0
  - encryption: 2.12.0
  - federatedfilesharing: 1.14.0
  - federation: 1.14.0
  - files: 1.19.0
  - files_pdfviewer: 2.5.0
  - files_rightclick: 1.3.0
  - files_sharing: 1.16.0
  - files_trashbin: 1.14.0
  - files_versions: 1.17.0
  - files_videoplayer: 1.13.0
  - firstrunwizard: 2.13.0
  - logreader: 2.9.0
  - lookup_server_connector: 1.12.0
  - mail: 1.11.7
  - nextcloud_announcements: 1.13.0
  - notifications: 2.12.0
  - oauth2: 1.12.0
  - password_policy: 1.14.0
  - photos: 1.6.0
  - privacy: 1.8.0
  - provisioning_api: 1.14.0
  - recommendations: 1.3.0
  - serverinfo: 1.14.0
  - settings: 1.6.0
  - sharebymail: 1.14.0
  - support: 1.7.0
  - survey_client: 1.12.0
  - systemtags: 1.14.0
  - text: 3.5.1
  - theming: 1.15.0
  - twofactor_backupcodes: 1.13.0
  - updatenotification: 1.14.0
  - user_status: 1.4.0
  - viewer: 1.8.0
  - weather_status: 1.4.0
  - workflowengine: 2.6.0
Disabled:
  - admin_audit
  - files_external
  - user_ldap
```

### Nextcloud Signing status

```shell
No errors have been found.
```

### Nextcloud Logs

```shell
{"reqId":"fFReINqBEr82jMCsGSv7","level":2,"time":"2022-02-08T19:27:57+00:00","remoteAddr":"87.155.22.20","user":"p609512","app":"ocs_api","method":"POST","url":"/ocs/v2.php/cloud
/users","message":"Failed addUser attempt with hint exception.","userAgent":"Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:96.0) Gecko/20100101 Firefox/96.0","version":"23.0.1.2","e
xception":{"Exception":"OC\\HintException","Message":"Password is present in compromised password list. Please choose a different password.","Code":0,"Trace":[{"file":"/html/next
cloud/apps/password_policy/lib/Listener/ValidatePasswordPolicyEventListener.php","line":46,"function":"validate","class":"OCA\\Password_Policy\\PasswordValidator","type":"->","ar
gs":["*** sensitive parameter replaced ***"]},{"file":"/html/nextcloud/lib/private/EventDispatcher/ServiceEventListener.php","line":87,"function":"handle","class":"OCA\\Password_
Policy\\Listener\\ValidatePasswordPolicyEventListener","type":"->","args":[{"__class__":"OCP\\Security\\Events\\ValidatePasswordPolicyEvent"}]},{"file":"/html/nextcloud/3rdparty/
symfony/event-dispatcher/EventDispatcher.php","line":251,"function":"__invoke","class":"OC\\EventDispatcher\\ServiceEventListener","type":"->","args":[{"__class__":"OCP\\Security
\\Events\\ValidatePasswordPolicyEvent"},"OCP\\Security\\Events\\ValidatePasswordPolicyEvent",{"__class__":"Symfony\\Component\\EventDispatcher\\EventDispatcher"}]},{"file":"/html
/nextcloud/3rdparty/symfony/event-dispatcher/EventDispatcher.php","line":73,"function":"callListeners","class":"Symfony\\Component\\EventDispatcher\\EventDispatcher","type":"->",
"args":[[{"__class__":"Closure"}],"OCP\\Security\\Events\\ValidatePasswordPolicyEvent",{"__class__":"OCP\\Security\\Events\\ValidatePasswordPolicyEvent"}]},{"file":"/html/nextclo
ud/lib/private/EventDispatcher/EventDispatcher.php","line":88,"function":"dispatch","class":"Symfony\\Component\\EventDispatcher\\EventDispatcher","type":"->","args":[{"__class__
":"OCP\\Security\\Events\\ValidatePasswordPolicyEvent"},"OCP\\Security\\Events\\ValidatePasswordPolicyEvent"]},{"file":"/html/nextcloud/lib/private/EventDispatcher/EventDispatche
r.php","line":100,"function":"dispatch","class":"OC\\EventDispatcher\\EventDispatcher","type":"->","args":["OCP\\Security\\Events\\ValidatePasswordPolicyEvent",{"__class__":"OCP\
\Security\\Events\\ValidatePasswordPolicyEvent"}]},{"file":"/html/nextcloud/lib/private/User/Database.php","line":123,"function":"dispatchTyped","class":"OC\\EventDispatcher\\Eve
ntDispatcher","type":"->","args":[{"__class__":"OCP\\Security\\Events\\ValidatePasswordPolicyEvent"}]},{"file":"/html/nextcloud/lib/private/User/Manager.php","line":463,"function
":"createUser","class":"OC\\User\\Database","type":"->","args":["*** sensitive parameter replaced ***","*** sensitive parameter replaced ***"]},{"file":"/html/nextcloud/lib/priva
te/User/Manager.php","line":407,"function":"createUserFromBackend","class":"OC\\User\\Manager","type":"->","args":["*** sensitive parameter replaced ***","*** sensitive parameter
 replaced ***",{"__class__":"OC\\User\\Database"}]},{"file":"/html/nextcloud/apps/provisioning_api/lib/Controller/UsersController.php","line":417,"function":"createUser","class":
"OC\\User\\Manager","type":"->","args":["*** sensitive parameter replaced ***","*** sensitive parameter replaced ***"]},{"file":"/html/nextcloud/lib/private/AppFramework/Http/Dis
patcher.php","line":217,"function":"addUser","class":"OCA\\Provisioning_API\\Controller\\UsersController","type":"->","args":["*** sensitive parameters replaced ***"]},{"file":"/
html/nextcloud/lib/private/AppFramework/Http/Dispatcher.php","line":126,"function":"executeController","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->","args":[{"__class_
_":"OCA\\Provisioning_API\\Controller\\UsersController"},"addUser"]},{"file":"/html/nextcloud/lib/private/AppFramework/App.php","line":157,"function":"dispatch","class":"OC\\AppF
ramework\\Http\\Dispatcher","type":"->","args":[{"__class__":"OCA\\Provisioning_API\\Controller\\UsersController"},"addUser"]},{"file":"/html/nextcloud/lib/private/Route/Router.p
hp","line":302,"function":"main","class":"OC\\AppFramework\\App","type":"::","args":["OCA\\Provisioning_API\\Controller\\UsersController","addUser",{"__class__":"OC\\AppFramework
\\DependencyInjection\\DIContainer"},{"_route":"ocs.provisioning_api.Users.addUser"}]},{"file":"/html/nextcloud/ocs/v1.php","line":62,"function":"match","class":"OC\\Route\\Route
r","type":"->","args":["/ocsapp/cloud/users"]},{"file":"/html/nextcloud/ocs/v2.php","line":23,"args":["/html/nextcloud/ocs/v1.php"],"function":"require_once"}],"File":"/html/next
cloud/apps/password_policy/lib/PasswordValidator.php","Line":87,"Hint":"Das Passwort ist in der Liste der kompromittierten Passw\u00f6rter enthalten. Bitte w\u00e4hle ein anderes
 Passwort.","CustomMessage":"Failed addUser attempt with hint exception."}}
```

**Comment by akh1** [comment]
Can confirm same issue,

finally restore from backup and avoid beta

webpack.modules.js - can be deleted safely
dist - not ease.

dist before (in ver 23) was located in ./core/js
so manually renaming file (from dist) and moving them in ./core/js  (not ./core/js/dist)  - will help resolve and upgrade
but again .... after next update - problem will rise again. so  i go back to ver23

**Comment by PVince81** [comment]
thanks for the report

you need to delete these two files before running updater.phar again

this is a known issue that will be ironed out for RC1

ref: https://github.com/nextcloud/updater/pull/412

**Comment by ChristophWurst** [comment]
I've compared the files. the rc3 and final .tar.bz2 contains the latest updater.phar from git. On my personal instance (rc3) the updater.phar is different.

This could mean that the updater was not replaced during the upgrade to RC3.

**Comment by blizzz** [comment]
> I've compared the files. the rc3 and final .tar.bz2 contains the latest updater.phar from git. On my personal instance (rc3) the updater.phar is different.
>
> This could mean that the updater was not replaced during the upgrade to RC3.

how did you upgrade?

**Comment by blizzz** [comment]
also, checked the updater.log?

**Comment by kaystrobach** [comment]
To avoid problems with migrations all the tables should be set to dynamic row format:

```
ALTER TABLE oc_mounts ROW_FORMAT=DYNAMIC;
```

Had to add this to get the migrations up and running, before the mariadb server just die and restarted during the updates

**Comment by kaystrobach** [comment]
Finally converted all the tables with the script from here:

* https://lxadm.com/MySQL:_changing_ROW_FORMAT_to_DYNAMIC_or_COMPRESSED

**Comment by ChristophWurst** [comment]
> @ChristophWurst did you double check your permissions on path/to/nc/updater[/updater.phar]?

```
-rw-r--r-- 1 www-data www-data 600659 Dez  1 09:08 /var/www/nextcloud/updater/updater.phar
```

```
2022-05-03T14:15:37+0200 S30ixZdZf2 [info] request to updater
2022-05-03T14:15:37+0200 S30ixZdZf2 [info] currentStep()
2022-05-03T14:15:37+0200 S30ixZdZf2 [info] show HTML page
2022-05-03T14:15:37+0200 S30ixZdZf2 [info] current version: 24.0.0 RC3 build time: 2022-04-28T12:37:46+00:00 04055e79c0295baffd912801cba8d915b65f297f
2022-05-03T14:15:37+0200 S30ixZdZf2 [info] getUpdateServerResponse()
2022-05-03T14:15:37+0200 S30ixZdZf2 [info] updaterServer: https://updates.nextcloud.com/internal_updater_server/
2022-05-03T14:15:37+0200 S30ixZdZf2 [info] releaseChannel: beta
2022-05-03T14:15:37+0200 S30ixZdZf2 [info] internal version: 24.0.0.11
2022-05-03T14:15:37+0200 S30ixZdZf2 [info] updateURL: https://updates.nextcloud.com/internal_updater_server/?version=24x0x0x11xxxbetaxx2022-04-28T12%3A37%3A46%2B00%3A00+04055e79c0295baffd912801cba8d915b65f297fx7x4x27
2022-05-03T14:15:37+0200 S30ixZdZf2 [info] getUpdateServerResponse response: Array
2022-05-03T14:15:37+0200 S30ixZdZf2 [info] checkForUpdate() Array
2022-05-03T14:15:37+0200 S30ixZdZf2 [info] getChangelogURL()
2022-05-03T14:15:37+0200 S30ixZdZf2 [info] end of checkForUpdate() Update to Nextcloud 24.0.0 available. (channel: "beta")<br /><span class="light">Following file will be downloaded automatically:</span> <code class="light">https://download.nextcloud.com/server/releases/nextcloud-24.0.0.zip</code><br /><a class="external_link" href="https://nextcloud.com/changelog/#24-0-0" target="_blank" rel="noreferrer noopener">Open changelog ↗</a>
2022-05-03T14:15:38+0200 sSHJwOUYhu [info] request to updater
2022-05-03T14:15:38+0200 sSHJwOUYhu [info] currentStep()
2022-05-03T14:15:38+0200 sSHJwOUYhu [info] POST request for step "1"
2022-05-03T14:15:38+0200 sSHJwOUYhu [info] startStep("1")
2022-05-03T14:15:38+0200 sSHJwOUYhu [info] checkForExpectedFilesAndFolders()
2022-05-03T14:15:38+0200 sSHJwOUYhu [error] POST request failed with UpdateException
2022-05-03T14:15:38+0200 sSHJwOUYhu [error] Exception: UpdateException
2022-05-03T14:15:38+0200 sSHJwOUYhu [info] rollbackChanges("1")
2022-05-03T14:15:38+0200 sSHJwOUYhu [info] unlink .step
2022-05-03T14:15:38+0200 sSHJwOUYhu [info] end of  rollbackChanges()
```

**Comment by schmidtvolker** [comment]
> Can confirm same issue,
> 
> finally restore from backup and avoid beta
> 
> webpack.modules.js - can be deleted safely dist - not ease.
> 
> dist before (in ver 23) was located in ./core/js so manually renaming file (from dist) and moving them in ./core/js (not ./core/js/dist) - will help resolve and upgrade but again .... after next update - problem will rise again. so i go back to ver23
> 
> can confirm that problem exist if upgrade from 23 to 24b1 and also from 24b1 to 24b2 and from 23 to 24b2

Can confirm same issue 24b1 to 24final

**Comment by blizzz** [comment]
y

> > Can confirm same issue,
> > finally restore from backup and avoid beta
> > webpack.modules.js - can be deleted safely dist - not ease.
> > dist before (in ver 23) was located in ./core/js so manually renaming file (from dist) and moving them in ./core/js (not ./core/js/dist) - will help resolve and upgrade but again .... after next update - problem will rise again. so i go back to ver23
> > can confirm that problem exist if upgrade from 23 to 24b1 and also from 24b1 to 24b2 and from 23 to 24b2
> 
> Can confirm same issue 24b1 to 24final

Beta1 was having the issue, the updater of beta 3 has  (or latest rc1, but should be beta3) the fix.

**Comment by blizzz** [comment]
@ChristophWurst nothing striking in the log file… do you have another idea, why the updater was not replaced on your sys? on three other instances, mixed web and cli installer, i did not see the issue and the updaters were replaced correctly.

**Comment by schmidtvolker** [comment]
> > > Can confirm same issue,
> > > finally restore from backup and avoid beta
> > > webpack.modules.js - can be deleted safely dist - not ease.
> > > dist before (in ver 23) was located in ./core/js so manually renaming file (from dist) and moving them in ./core/js (not ./core/js/dist) - will help resolve and upgrade but again .... after next update - problem will rise again. so i go back to ver23
> > > can confirm that problem exist if upgrade from 23 to 24b1 and also from 24b1 to 24b2 and from 23 to 24b2
> > 
> > 
> > Can confirm same issue 24b1 to 24final
> 
> Beta1 was having the issue, the updater of beta 3 has (or latest rc1, but should be beta3) the fix.


I just tried an update with this results:

Current version is 24.0.0 beta 1.

Update to Nextcloud 24.0.0 available. (channel: "beta")
Following file will be downloaded automatically: https://download.nextcloud.com/server/releases/nextcloud-24.0.0.zip
Open changelog ↗

Steps that will be executed:
[ ] Check for expected files
[ ] Check for write permissions
[ ] Create backup
[ ] Downloading
[ ] Verify integrity
[ ] Extracting
[ ] Enable maintenance mode
[ ] Replace entry points
[ ] Delete old files
[ ] Move new files in place
[ ] Done

Start update? [y/N] y

Info: Pressing Ctrl-C will finish the currently running step and then stops the updater.

[✘] Check for expected files failed
The following extra files have been found:
    dist

Update failed. To resume or retry just execute the updater again.
  - installed: true
  - version: 24.0.0.6
  - versionstring: 24.0.0 beta 1
  - edition: 
  - maintenance: false
  - needsDbUpgrade: false
  - productname: Nextcloud
  - extendedSupport: false
Nextcloud 24.0.0 beta 1

**Comment by ChristophWurst** [comment]
> @ChristophWurst nothing striking in the log file… do you have another idea, why the updater was not replaced on your sys? on three other instances, mixed web and cli installer, i did not see the issue and the updaters were replaced correctly.

The instance was on `internal version: 23.0.0.10` before the jump to 24 RC3 if that helps.

**Comment by schmidtvolker** [comment]
@blizzz thx :) it works 👍

**Comment by blizzz** [comment]
> I have backups of the directory. Is there anything important info I could find in a time travel?

Could you e.g. dockerize and reprodce it?

**Comment by JulianFP** [comment]
I can confirm that issue for an update from 4.0.0 stable to 4.0.1 stable as well. I had never any beta versions installed. The dist directory is also missing in updater.phar. Permission were correctly set.
![Screenshot from 2022-05-23 18-21-06](https://user-images.githubusercontent.com/70963316/169867856-6e75d7e4-27a8-48d8-ab3b-641da020f156.png)
![Screenshot from 2022-05-23 12-16-32](https://user-images.githubusercontent.com/70963316/169867863-eb17433d-e1bd-4c8d-b43e-52f569adeeea.png)

**Comment by dirks** [comment]
Same from 24.0.1 to 24.0.2 update. As a data point `updater.phar` on that instance was from October 2021. If more specific information on the `updater.phar` could help to pin the issue down, let me know and I dig it from an old snapshot.

Workaround [manual upgrade](https://docs.nextcloud.com/server/24/admin_manual/maintenance/manual_upgrade.html), which brings a new `update.phar` from archive

**Comment by dirks** [comment]
> > Workaround [manual upgrade](https://docs.nextcloud.com/server/24/admin_manual/maintenance/manual_upgrade.html), which includes a new `updater.phar` from archive
> 
> Another maybe more easy workaround would be to just replace the updater.phar file from a downloaded .tar of your Nextcloud version. Then you can update your Nextcloud instance with the new updater file.

Sure it is easier, but I am not involved enough in the project to know, if there are any dependencies to the `updater.phar` which might suffering from the same bug (i.e. have not updated during previous updates). That is why I lean towards the alternative and documented upgrade path which should circumvent the problem completely. Your mileage my vary. :balance_scale:

**Comment by blizzz** [comment]
what are the permissions on the updater.phar for those affected?

**Comment by ChristophWurst** [comment]
```rw-r--r-- 1 www-data www-data``` (Debian)

**Comment by dhzl84** [comment]
`-rw-r--r-- 1 root root` (Debian 10)

**Comment by blizzz** [comment]
> `-rw-r--r-- 1 root root` (Debian 10)

The web user does not have permissions to replace them.

**Comment by dhzl84** [comment]
Yes, no clue where it came from, changed it to www-data and will keep an eye on it.

**Comment by akh1** [comment]
My guess - problem exist if SELinux in enforcing mode
try  setenforce 0 in shell BEFORE update

**Comment by dhzl84** [comment]
> Yes, no clue where it came from, changed it to www-data and will keep an eye on it.

24.0.3 to 24.04 update shows same issue, even with `www-data` as owner of `dist`

> My guess - problem exist if SELinux in enforcing mode
try setenforce 0 in shell BEFORE update

SELinux is not installed

I even added the `dist` folder manually in`updater.phar` but no success.
<img width="331" alt="image" src="https://user-images.githubusercontent.com/5675570/184333696-b0b1cd57-4500-40f4-9aa8-6eb50aabdc18.png">

**Comment by AlexanderKorff** [comment]
> I even added the `dist` folder manually in`updater.phar` but no success.

Adding dist to the updater/index.php file solved the issue for me.

```
  private function getExpectedElementsList() {
                $expected = [
                        // Generic
                        '.',
                        '..',
                        // Folders
                        '.well-known',
                        '3rdparty',
                        'apps',
                        'config',
                        'core',
                        'data',
                        'dist',
                        ...
```

**Comment by dhzl84** [comment]
> 

Confirmed for 24.0.4 to 24.0.5, thanks!

**Comment by Srokap** [comment]
Same problem out of the blue when upgrading from 24.0.5 to 24.0.7

`updater/index.php` was missing `dist` in expected elements list.

**Comment by Fantu** [comment]
I upgraded from 23.0.12 to 24.0.8 without issues and now trying to upgrade to 25.0.2 shows this error, I also manually changed updater/index.php

**Comment by rasos** [comment]
Same issue when upgrading from 24.0.7 to 24.0.9 with `updater.phar`: _The following extra files have been found:  dist  . Update failed._

**Comment by rasos** [comment]
This must have been a packaging issue for some 24.0.x releases. `updater.phar` had a date in February 2022 on two  nextcloud instances I maintain  (no beta). It seems the `updater` folder was not updated when upgrading to NC 24. Some packaother NC 24 instances I maintain were okay. Maybe the packaging procedure has been fixed at some point.

I fixed it by downloading that release, and replacing `updater.phar`. 

```
wget https://download.nextcloud.com/server/releases/nextcloud-24.0.6.zip
unzip nextcloud-24.0.6.zip
cp /nextcloud/updater/updater.phar /var/www/nextcloud/updater/updater.phar
sudo -u www-data /usr/bin/php /var/www/nextcloud/updater/updater.phar
cd ~
rmdir -r nextcloud
rm nextcloud-24.0.6.zip
```

**Comment by szaimen** [comment]
Hi, please update to 25.0.7 or better 26.0.2 and report back if it fixes the issue. Thank you!

My goal is to add a label like e.g. 26-feedback to this ticket of an up-to-date major Nextcloud version where the bug could be reproduced. However this is not going to work without your help. So thanks for all your effort!

If you don't manage to reproduce the issue in time and the issue gets closed but you can reproduce the issue afterwards, feel free to create a new bug report with up-to-date information by following this link: https://github.com/nextcloud/server/issues/new?assignees=&labels=bug%2C0.+Needs+triage&template=BUG_REPORT.yml&title=%5BBug%5D%3A+

**Comment by antoneliasson** [comment]
I just updated from 23.0.12 to 24.0.12. updater/index.php and updater/updater.phar were not updated and thus caused this error when subsequently attempting to update to 25.0.12. `occ integrity:check-core` did not find any issue even if I manually modify updater/index.php and updater/updater.phar so apparently those files are not included in check-core.

The checksums of the two files match the ones in https://github.com/nextcloud/updater/tree/v23.0.12 and have modification dates from Dec 17 2022 which is probably when I upgraded to 23.0.12.

Fixed by manually overwriting the two files with the ones from nextcloud 24.0.12 similar to how rasos did it two comments above: https://github.com/nextcloud/server/issues/31891#issuecomment-1404079753

After the upgrade to 25.0.12 I still have updater/index.php and updater/updater.phar matching 24.0.12. So I think the problem that updater files not being updated is still present in Nextcloud 25.

If the issue remains after I get around to updating to Nextcloud 26 then I might make a new bug report.
